emacs: 25.3 → 26.1

matthewbauer opened the pull request:
###### Motivation for this change
New Emacs release!

Note: Macport still needs to be updated. Waiting on mituharu for release.

matthewbauer committed:
emacs: 25.3 → 26.1

matthewbauer committed:
emacs: readd version 25

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.41181&attempt_id=70a4afb7-4323-429e-8f78-d56bd19d29ed)

Attempted: emacs

<details><summary>Partial log (click to expand)</summary><p>

```
    reference to free variable `tramp-remote-path'
/nix/store/p8ylblab1cdybfl9z8kddncp5zdgiaml-emacs-26.1/share/emacs/site-lisp/site-start.el:37:18:Warning:
    assignment to free variable `tramp-remote-path'
post-installation fixup
gzipping man pages under /nix/store/p8ylblab1cdybfl9z8kddncp5zdgiaml-emacs-26.1/share/man/
strip is /nix/store/r5mcn9vqq80v5pfqp45d7290cis7dwp4-cctools-binutils-darwin/bin/strip
stripping (with command strip and flags -S) in /nix/store/p8ylblab1cdybfl9z8kddncp5zdgiaml-emacs-26.1/lib  /nix/store/p8ylblab1cdybfl9z8kddncp5zdgiaml-emacs-26.1/libexec  /nix/store/p8ylblab1cdybfl9z8kddncp5zdgiaml-emacs-26.1/bin
patching script interpreter paths in /nix/store/p8ylblab1cdybfl9z8kddncp5zdgiaml-emacs-26.1
/nix/store/p8ylblab1cdybfl9z8kddncp5zdgiaml-emacs-26.1/libexec/emacs/26.1/x86_64-apple-darwin16.3.0/rcs2log: interpreter directive changed from " /bin/sh" to "/nix/store/i9g6883w05pyj9aa4sl8hn945qyy6j53-bash-4.4-p19/bin/sh"
/nix/store/p8ylblab1cdybfl9z8kddncp5zdgiaml-emacs-26.1
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.41181&attempt_id=f59dd927-c3d0-4801-9dc4-eb3ade76a8cf)

Attempted: emacs

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/p8ylblab1cdybfl9z8kddncp5zdgiaml-emacs-26.1
```
</p></details>

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.41181&attempt_id=005eca24-357b-49b2-8a3e-164b119cbbdf)

Attempted: emacs

<details><summary>Partial log (click to expand)</summary><p>

```
   Compiling filetime v0.1.15
   Compiling build_helper v0.1.0 (file:///build/rustc-1.25.0-src/src/build_helper)
   Compiling rustc_binaryen v0.0.0 (file:///build/rustc-1.25.0-src/src/librustc_binaryen)
   Compiling rustc_llvm v0.0.0 (file:///build/rustc-1.25.0-src/src/librustc_llvm)
building of '/nix/store/04rc3rxknhf3gpxs0f4l37b6l7hh9kxi-rustc-1.25.0.drv' timed out after 3600 seconds
cannot build derivation '/nix/store/jpc25qranb2dz8vxwf1n6dg594ssi8yc-cargo-0.26.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/b46kvcjs5n1c0wyb7f8f6kyfwlhcdi6j-librsvg-2.42.4.drv': 2 dependencies couldn't be built
cannot build derivation '/nix/store/2cks397xgskkcf8n0llxgkhddsbxhlhk-hook.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/ixgxpgkqfrs6j5fpz4lgfd47aq1riiyj-emacs-26.1.drv': 2 dependencies couldn't be built
[31;1merror:[0m build of '/nix/store/ixgxpgkqfrs6j5fpz4lgfd47aq1riiyj-emacs-26.1.drv' failed
```
</p></details>

GrahamcOfBorg commented:
Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.41181&attempt_id=7e1bf0b1-cee8-4a3b-8a54-274e85f1dfde)

Attempted: emacs

<details><summary>Partial log (click to expand)</summary><p>

```
test [run-pass] run-pass/deriving-via-extension-hash-struct.rs ... ok
test [run-pass] run-pass/deriving-show.rs ... ok
test [run-pass] run-pass/deriving-via-extension-struct-empty.rs ... ok
test [run-pass] run-pass/deriving-show-2.rs ... ok
building of '/nix/store/iwv52k8g0r4brn2dpc10zhz7y40q245s-rustc-1.25.0.drv' timed out after 4000 seconds
cannot build derivation '/nix/store/m1w2n07q9js00hbxw8r9g96v9r6pfrfs-cargo-0.26.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/hs4ykgrrnkbdixyli7q9z5m5j7yxy7xa-librsvg-2.42.4.drv': 2 dependencies couldn't be built
cannot build derivation '/nix/store/07081isg4q5km0abr0kmy1f98y6xidgc-hook.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/nij5pfvpz4k8jl6mmlavjg6dgh6bql2m-emacs-26.1.drv': 2 dependencies couldn't be built
error: build of '/nix/store/nij5pfvpz4k8jl6mmlavjg6dgh6bql2m-emacs-26.1.drv' failed
```
</p></details>

matthewbauer committed:
emacs26: add some tweaks from jwiegley’s overlay

original src:
https://github.com/jwiegley/nix-config/blob/f85c560bf8165fecbbd6a28bc2ba0ebbafeb3cf3/overlays/10-emacs.nix

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.41181&attempt_id=d0a67bb3-cadf-431e-819d-cf4be5cb7e9d)

Attempted: emacs

<details><summary>Partial log (click to expand)</summary><p>

```
   Compiling backtrace v0.3.5
   Compiling proc_macro v0.0.0 (file:///build/rustc-1.25.0-src/src/libproc_macro)
   Compiling rustc_const_math v0.0.0 (file:///build/rustc-1.25.0-src/src/librustc_const_math)
   Compiling syntax_ext v0.0.0 (file:///build/rustc-1.25.0-src/src/libsyntax_ext)
building of '/nix/store/04rc3rxknhf3gpxs0f4l37b6l7hh9kxi-rustc-1.25.0.drv' timed out after 3600 seconds
cannot build derivation '/nix/store/jpc25qranb2dz8vxwf1n6dg594ssi8yc-cargo-0.26.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/b46kvcjs5n1c0wyb7f8f6kyfwlhcdi6j-librsvg-2.42.4.drv': 2 dependencies couldn't be built
cannot build derivation '/nix/store/2cks397xgskkcf8n0llxgkhddsbxhlhk-hook.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/ixgxpgkqfrs6j5fpz4lgfd47aq1riiyj-emacs-26.1.drv': 2 dependencies couldn't be built
[31;1merror:[0m build of '/nix/store/ixgxpgkqfrs6j5fpz4lgfd47aq1riiyj-emacs-26.1.drv' failed
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.41181&attempt_id=7a44f1b2-30bd-4f44-bf10-1430d4611995)

Attempted: emacs, emacs26

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/wyq3g9aa0xjq97vxir410h22hafvm1cn-emacs-26.1/share/emacs/site-lisp/site-start.el:37:18:Warning:
    assignment to free variable `tramp-remote-path'
post-installation fixup
gzipping man pages under /nix/store/wyq3g9aa0xjq97vxir410h22hafvm1cn-emacs-26.1/share/man/
strip is /nix/store/r5mcn9vqq80v5pfqp45d7290cis7dwp4-cctools-binutils-darwin/bin/strip
stripping (with command strip and flags -S) in /nix/store/wyq3g9aa0xjq97vxir410h22hafvm1cn-emacs-26.1/lib  /nix/store/wyq3g9aa0xjq97vxir410h22hafvm1cn-emacs-26.1/libexec  /nix/store/wyq3g9aa0xjq97vxir410h22hafvm1cn-emacs-26.1/bin
patching script interpreter paths in /nix/store/wyq3g9aa0xjq97vxir410h22hafvm1cn-emacs-26.1
/nix/store/wyq3g9aa0xjq97vxir410h22hafvm1cn-emacs-26.1/libexec/emacs/26.1/x86_64-apple-darwin17.3.0/rcs2log: interpreter directive changed from " /bin/sh" to "/nix/store/i9g6883w05pyj9aa4sl8hn945qyy6j53-bash-4.4-p19/bin/sh"
/nix/store/wyq3g9aa0xjq97vxir410h22hafvm1cn-emacs-26.1
/nix/store/wyq3g9aa0xjq97vxir410h22hafvm1cn-emacs-26.1
```
</p></details>

GrahamcOfBorg commented:
Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.41181&attempt_id=c448d8ca-56c8-4080-bd77-54220a804f26)

Attempted: emacs

<details><summary>Partial log (click to expand)</summary><p>

```
test [run-fail] run-fail/bug-811.rs ... ok
test [run-fail] run-fail/bug-2470-bounds-check-overflow.rs ... ok
test [run-fail] run-fail/call-fn-never-arg.rs ... ok
test [run-fail] run-fail/borrowck-local-borrow.rs ... ok
building of '/nix/store/iwv52k8g0r4brn2dpc10zhz7y40q245s-rustc-1.25.0.drv' timed out after 4000 seconds
cannot build derivation '/nix/store/m1w2n07q9js00hbxw8r9g96v9r6pfrfs-cargo-0.26.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/hs4ykgrrnkbdixyli7q9z5m5j7yxy7xa-librsvg-2.42.4.drv': 2 dependencies couldn't be built
cannot build derivation '/nix/store/07081isg4q5km0abr0kmy1f98y6xidgc-hook.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/nij5pfvpz4k8jl6mmlavjg6dgh6bql2m-emacs-26.1.drv': 2 dependencies couldn't be built
error: build of '/nix/store/nij5pfvpz4k8jl6mmlavjg6dgh6bql2m-emacs-26.1.drv' failed
```
</p></details>

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.41181&attempt_id=ea1cb70f-3ab0-4536-9a24-e3b972375487)

Attempted: emacs, emacs26

<details><summary>Partial log (click to expand)</summary><p>

```
   Compiling filetime v0.1.15
   Compiling build_helper v0.1.0 (file:///build/rustc-1.25.0-src/src/build_helper)
   Compiling rustc_binaryen v0.0.0 (file:///build/rustc-1.25.0-src/src/librustc_binaryen)
   Compiling rustc_llvm v0.0.0 (file:///build/rustc-1.25.0-src/src/librustc_llvm)
building of '/nix/store/04rc3rxknhf3gpxs0f4l37b6l7hh9kxi-rustc-1.25.0.drv' timed out after 3600 seconds
cannot build derivation '/nix/store/jpc25qranb2dz8vxwf1n6dg594ssi8yc-cargo-0.26.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/73906rjsar0659qnmyapfxxzw47fq5bg-librsvg-2.42.4.drv': 2 dependencies couldn't be built
cannot build derivation '/nix/store/kbg5w8hk1mfjkjlih7is2wqyifx1bnfs-hook.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/igkndy45khwngzy60ar6vn9cj1y0gfm7-emacs-26.1.drv': 2 dependencies couldn't be built
[31;1merror:[0m build of '/nix/store/igkndy45khwngzy60ar6vn9cj1y0gfm7-emacs-26.1.drv' failed
```
</p></details>

GrahamcOfBorg commented:
Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.41181&attempt_id=34ed2803-d6ae-49ed-9fba-33f63d622c32)

Attempted: emacs, emacs26

<details><summary>Partial log (click to expand)</summary><p>

```
   Compiling backtrace v0.3.5
   Compiling rustc_const_math v0.0.0 (file:///build/rustc-1.25.0-src/src/librustc_const_math)
   Compiling proc_macro v0.0.0 (file:///build/rustc-1.25.0-src/src/libproc_macro)
   Compiling syntax_ext v0.0.0 (file:///build/rustc-1.25.0-src/src/libsyntax_ext)
building of '/nix/store/iwv52k8g0r4brn2dpc10zhz7y40q245s-rustc-1.25.0.drv' timed out after 4000 seconds
cannot build derivation '/nix/store/m1w2n07q9js00hbxw8r9g96v9r6pfrfs-cargo-0.26.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/qml59i5m9rcy4na5fsavvjn6j28dby07-librsvg-2.42.4.drv': 2 dependencies couldn't be built
cannot build derivation '/nix/store/5756v6wrjaiz304v44xyxfqv1rarqgsd-hook.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/l2lck83dfqzg13c6x4306p2jp11f3asm-emacs-26.1.drv': 2 dependencies couldn't be built
error: build of '/nix/store/l2lck83dfqzg13c6x4306p2jp11f3asm-emacs-26.1.drv' failed
```
</p></details>
